com.unity.multiplayer.mlapi/Runtime/Timing/ClientNetworkTimeProvider.cs

```diff
@@ -0,0 +1,121 @@
+using UnityEngine;
+
+namespace MLAPI.Timing
+{
+    /// <summary>
+    /// A <see cref="INetworkTimeProvider"/> which can be used on a client machine to control  and server time based on network conditions.
+    /// The <see cref="ClientNetworkTimeProvider"/> speeds up or slows down network time when necessary to adjust for RTT changes.
+    /// <see cref="ClientNetworkTimeProvider"/> is one of many ways to implement a network time provider for a client.
+    /// If the existing properties are not sufficient one can write a custom provider by implementing the <see cref="INetworkTimeProvider"/> interface.
+    /// </summary>
+    public class ClientNetworkTimeProvider : INetworkTimeProvider
+    {
+        private INetworkStats m_NetworkStats;
+
+        /// <summary>
+        /// Gets or sets the initial RTT in seconds used when starting a client.
+        /// This value will just be used for the initial time values after that the <see cref="m_NetworkStats"/> will provide the RTT.
+        /// </summary>
+        public float StartRtt { get; set; } = 150f / 1000f; // 150ms
+
+        /// <summary>
+        /// Gets or sets the ratio at which the <see cref="ClientNetworkTimeProvider"/> speeds up or slows down time.
+        /// </summary>
+        public float AdjustmentRatio { get; set; } = 0.01f;
+
+        /// <summary>
+        /// Gets or sets a threshold in seconds used to force a hard catchup of network time.
+        /// </summary>
+        public float HardResetThreshold { get; set; } = 200f / 1000f; // 200ms
+
+        /// <summary>
+        /// Gets the current time scale for server time.
+        /// </summary>
+        public float ServerTimeScale { get; private set; } = 1f;
+
+        /// <summary>
+        /// Gets the current time scale for local time.
+        /// </summary>
+        public float LocalTimeScale { get; private set; } = 1f;
+
+        /// <summary>
+        /// Gets or sets the amount of time in seconds the server should buffer incoming client messages.
```

Comment:
It might help to indicate whether some of these vars are things users would set.  And actually I'm a little confused, I figured I would find code in here that keeps adjusting TargetClientBufferTime based on the server RTT.  Or is it something also that users would set?  Or maybe for overriding the default to get started but then the system adjusts it in real time?